app/code/community/Ebizmarts/MailChimp/Model/ProcessWebhook.php

```diff
@@ -272,8 +277,57 @@ public function _profile(array $data)
                      */
                     $this->_addSubscriberData($subscriber, $fname, $lname, $email, $listId);
                 }
+
+                if (isset($data['merges']['GROUPINGS'])) {
+                    $this->_processGroupsData($data['merges']['GROUPINGS'], $subscriber, $listId, true);
+                }
+            }
+        }
+    }
+
+    public function _processGroupsData($groupings, $customerData, $listId, $isDataSubscriber = false)
+    {
+        $groups = array();
+        $helper = $this->getHelper();
+        $dateHelper = $this->getDateHelper();
+
+        $customerEmail = $customerData->getEmail();
+        $subscriber = $this->getSubscriberModel()->loadByEmail($customerEmail);
+        $storeId = $subscriber->getStoreId();
+
+        $api = $helper->getApi($storeId);
+        $apiInterests = $api->getLists()->getInterestCategory()->getInterests();
+
+        foreach ($groupings as $grouping) {
+            $interests = $apiInterests->getAll($listId, $grouping['unique_id']);
+
+            $groupsSave = array();
+            foreach ($interests['interests'] as $mcGroup) {
+                if (strpos($grouping['groups'], $mcGroup['name']) !== false) {
+                    $groupsSave [$mcGroup['id']] = $mcGroup['id'];
+                }
             }
+            $groups [$grouping['unique_id']]= $groupsSave;
         }
+
+        if (!$isDataSubscriber) {
+            $customerId = $customerData->getId();
+        } else {
+            $customerId = $customerData->getCustomerId();
+        }
+
+        $interestGroup = $this->getInterestGroupModel();
+
+        $subscriberId = $subscriber->getSubscriberId();
+        $interestGroup->getByRelatedIdStoreId($customerId, $subscriberId, $storeId);
+        $encodedGroups = $helper->arrayEncode($groups);
+
+        $interestGroup->setGroupdata($encodedGroups);
+        $interestGroup->setSubscriberId($subscriberId);
+        $interestGroup->setCustomerId($customerId);
+        $interestGroup->setStoreId($storeId);
+        $interestGroup->setUpdatedAt($dateHelper->getCurrentDateTime());
+        $interestGroup->save();
     }
```

Comment:
Please move function to a new class at app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers/InterestGroupHandle.php